sweep/fee_bumper.go

```diff
@@ -263,9 +280,15 @@ func (b *BumpResult) Validate() error {
 		return fmt.Errorf("%w: nil replacing tx", ErrInvalidBumpResult)
 	}
 
-	// If it's a failed event, it must have an error.
-	if b.Event == TxFailed && b.Err == nil {
-		return fmt.Errorf("%w: nil error", ErrInvalidBumpResult)
+	// If it's a failed or fatal event, it must have an error.
+	if b.Err == nil {
+		if b.Event == TxFailed {
```

Comment:
feels like this check should be the other way around?

```
// If it's a failed or fatal event, it must have an error.
	if b.Event == TxFatal || b.Event == TxFailed {
		if b.Err == nil {
			return fmt.Errorf("%w: nil error", ErrInvalidBumpResult)
		}
	}
```

Reply:
updated